sdk/resourcemanager/azure-resourcemanager-storage/src/main/java/com/azure/resourcemanager/storage/models/StorageAccount.java

```diff
@@ -381,6 +408,44 @@ interface WithEncryption {
              * @return the next stage of storage account definition
              */
             WithCreate withEncryptionKeyFromKeyVault(String keyVaultUri, String keyName, String keyVersion);
+
+            /**
+             * Specifies the KeyVault key to be used as key for encryption and the user-assigned identity to access the KeyVault,
+             * meaning this Storage Account will be encrypted with Customer-managed keys.
+             * If the provided user-assigned identity is null, this method is equivalent to {@link WithEncryption#withEncryptionKeyFromKeyVault(String, String, String)},
+             * meaning it'll use the system-assigned identity to access the KeyVault.
```

Comment:
Personally a bit unsure about whether we should "require" user sets a non-null identity.

On one hand it better to require non-null given the API shape.

On the other hand I can think of some case that putting a nullable identity would be simpler than conditioning (and fluent style is hard to do a `if else`).

So maybe keep this.

Reply:
I get your point. The meaning of putting a null identity here is a bit vague to the user, even with the javadocs. User could accidentally provide a null identity and the behavior may surprise them. We should require a non-null identity.